### ISSUE TITLE
Selector dusk with html tag - An invalid or illegal selector was specified

### ISSUE THREAD

**Issue by dborovsky** [laravel/dusk#410]
Hi

I added dusk selector to html tag button:

```<button dusk="register-button" class="btn btn-primary btn-block" type="submit" name="Register">Register</button>```

My test:

```
public function testUserCanSuccessSignup()
    {
        $this->browse(function ($browser) {
            $browser->visit('/')
                ->clickLink('register') 
                ->assertSee('Register')
                ->type('name', 'Joeeeeeeeeeeee') 
                ->type('email', 'joe@example.com')
                ->type('password', '123456')
                ->type('password_confirmation', '123456')
                ->click('@register-button')
                ->assertPathIs('/');
        });
    }
```

When running Dusk tests the following exception is thrown:

Facebook\WebDriver\Exception\InvalidSelectorException: invalid selector: An invalid or illegal selector was specified

**Comment by unglud** [comment]
Try to add
`dd($this->resolver->format($selector));`
to `vendor/laravel/dusk/src/Concerns/InteractsWithElements.php:43`
like this

```
public function click($selector)
    {
        dd($this->resolver->format($selector));
        $this->resolver->findOrFail($selector)->click();

        return $this;
    }
```
then run test, you should see `body [dusk="register-button"]` in console. But I guess you will se something different.

**Comment by dborovsky** [comment]
I got in console `"body @register-button"`

Hm... Any idee?

**Comment by unglud** [comment]
This is super wrong I would say. Try remove `vendor` folder completely and run `composer update` again (not `install`). I pretty sure problem will be solved. If not, try to trace why `vendor/laravel/dusk/src/ElementResolver.php:371` line didn't work correctly for you. 

```php
if (starts_with($selector, '@') && $selector === $originalSelector) {
    $selector = '[dusk="'.explode('@', $selector)[1].'"]';
}
```

**Comment by dborovsky** [comment]
Yea, good. I updated and error was solved.  The reason was in the version dusk. I used v 1.0. Thank you.

**Comment by dbudwin** [comment]
I'm experiencing the same issue.  I have a button that I am attempting to click that has a `dusk` tag in it.  When I added `dd($this->resolver->format($selector));` as suggested above by @unglud, I got a similar result as @dborovsky did.  My result says `"body @addDeviceButton"`.  I deleted my `vendors` folder and ran `composer update` as suggested which gave me Dusk 1.1.1.  When I reran the test, I got the exact same error as before.  I closed and restarted my Chrome driver and PHPStorm after updating.

_My test_:
```
public function testDevices_ClickAddDeviceButton_OpensAddDeviceModal(): void
{
    $this->browse(function (Browser $browser) {
        $user = $this->createUser();
        $browser->loginAs($user)->visit('/devices')->click('@addDeviceButton')->assertSee('Device Name');
    });
}
```

_My partial HTML_:
```
<div class="panel-footer">
    <button dusk="addDeviceButton" type="button" class="btn btn-default" aria-label="Add Device" data-toggle="modal" data-target="#addDeviceModal">
        <span class="glyphicon glyphicon-plus"></span> Add Device
    </button>
    @include('partials.add-device-modal')
    @include('partials.edit-device-modal')
</div>
```

Looking into the Dusk code as mentioned above, it seems to intentionally prepend `body` to the selector name ([which is different than what the documentation suggests](https://laravel.com/docs/5.5/dusk#interacting-with-elements)).

_Screenshot of my debugger showing how it's formatting my selector to prepend `body`_
<img width="899" alt="screen shot 2018-01-06 at 11 20 28" src="https://user-images.githubusercontent.com/15161181/34641555-00d17ec2-f2d4-11e7-871e-8287e1bd2728.png">
